From b91879841747470182c517b5990f90c8de3291a8 Mon Sep 17 00:00:00 2001
From: Zheng Zengkai <zhengzengkai@huawei.com>
Date: Fri, 16 Dec 2022 17:40:25 +0800
Subject: [PATCH] efi/libstub: Fix build error in efi-stub.c for riscv
MIME-Version: 1.0
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b91879841747470182c517b5990f90c8de3291a8
Modified-by-SEL: No

Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I66G0M

-----------------------------------------------

Commit 36a1a8916de5 ("efi/loongarch: Add efistub booting support")
introduced the following macro definition for riscv in file
drivers/firmware/efi/libstub/efi-stub.c

 # define EFI_RT_VIRTUAL_LIMIT  DEFAULT_MAP_WINDOW_64
 # define EFI_RT_VIRTUAL_LIMIT  TASK_SIZE_MIN
 # define EFI_RT_VIRTUAL_LIMIT  TASK_SIZE

but related patches have not been merged for riscv, like
e8a62cc26ddf ("riscv: Implement sv48 support"),
01abdfeac81b ("riscv: compat: Support TASK_SIZE for compat mode"),

Causing build error in riscv:

build failed: riscv, allmodconfig

<line too long ...>
n this function); did you mean ‘TASK_SIZE_MAX’?
  # define EFI_RT_VIRTUAL_LIMIT TASK_SIZE_MIN
                                ^
drivers/firmware/efi/libstub/efi-stub.c:289:31: note: in expansion of macro ‘EFI_RT_VIRTUAL_LIMIT’
   static const u64 headroom = EFI_RT_VIRTUAL_LIMIT -
                               ^~~~~~~~~~~~~~~~~~~~
drivers/firmware/efi/libstub/efi-stub.c:44:32: note: each undeclared identifier is reported only once for each function it appears in
  # define EFI_RT_VIRTUAL_LIMIT TASK_SIZE_MIN
                                ^
drivers/firmware/efi/libstub/efi-stub.c:289:31: note: in expansion of macro ‘EFI_RT_VIRTUAL_LIMIT’
   static const u64 headroom = EFI_RT_VIRTUAL_LIMIT -
                               ^~~~~~~~~~~~~~~~~~~~
make[4]: *** [drivers/firmware/efi/libstub/efi-stub.o] Error 1
make[3]: *** [drivers/firmware/efi/libstub] Error 2
make[2]: *** [drivers/firmware/efi] Error 2
make[1]: *** [drivers/firmware] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [drivers] Error 2

Fix it by removing the CONFIG_RISCV condition for EFI_RT_VIRTUAL_LIMIT definition.

Fixes: 36a1a8916de5 ("efi/loongarch: Add efistub booting support")
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Reviewed-by: Wei Li <liwei391@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/firmware/efi/libstub/efi-stub.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/firmware/efi/libstub/efi-stub.c b/drivers/firmware/efi/libstub/efi-stub.c
index 8e0f64b3db69..96129f0fc60e 100644
--- a/drivers/firmware/efi/libstub/efi-stub.c
+++ b/drivers/firmware/efi/libstub/efi-stub.c
@@ -40,7 +40,7 @@
 
 #ifdef CONFIG_ARM64
 # define EFI_RT_VIRTUAL_LIMIT	DEFAULT_MAP_WINDOW_64
-#elif defined(CONFIG_RISCV) || defined(CONFIG_LOONGARCH)
+#elif defined(CONFIG_LOONGARCH)
  # define EFI_RT_VIRTUAL_LIMIT	TASK_SIZE_MIN
 #else
 # define EFI_RT_VIRTUAL_LIMIT	TASK_SIZE
-- 
2.33.0

